### DISCUSSION TITLE
Сортировка с помощью кнопки

### DISCUSSION THREAD

**Discussion by horyachka** [totumonline/totum-ru-issues-and-discussions#76] [Как это сделать?] [answered]
Добрый день. Подскажите пожалуйста как правильно написать скрипт для сортировки таблицы с помощью кнопки.
Пример таблицы:

![image](https://user-images.githubusercontent.com/91678200/154680121-f0108a8f-88f2-4db3-9e4a-2810b3128f4d.png)

Код действия:
![image](https://user-images.githubusercontent.com/91678200/154681342-c10d253e-a54a-4503-9c2a-4eec53af4770.png)

= :listSort(list: $selectlist; type: "number"; direction: "desc")
selectlist: selectList(table: $#ntn; field: 'kolichestvo')

Нажимаю кнопку и сортировка не происходит. 
Сортировка нужна как в простой так и временной таблице.
Спасибо!

**Comment by erpic-ru** [comment, marked as answer]
Добрый день! Такой код работать не будет. Он не оказывает воздействия на табличную часть. Т.к. только берет и сортирует. Тут нужно продолжение в виде set: или insert:

Вот в этом видео есть вариант решения, возможно, вам подойдет:
https://youtu.be/BDcd4a4AJ2c?t=261

Из очевидного: сортировку и фильтрацию можно включать по каждой колонке. Либо поменять поле сортировки в настройках таблицы.

**Reply by ttm-alexeypolunin** [reply]
**Важное замечание** — что сортировка по умолчанию отрабатывает только при загрузке таблицы!

Если необходимо, что бы именно по нажатию кнопки происходила сортировка, то можно использовать процессную переменную — которая живет в течении транзакции.

В этом случае мы в кнопке ставим переменную:

```
= : procVar(name: "order"; value: true)
```

А в коде форматирования таблицы пишем:

```
f1=: if(condition: $@order = true; then: $order)

	order: setRowsOrder(ids: $sort[[id]])
	
		sort: listSort(list: $sel; key: "order_field" asc num)
	
			sel: selectRowList(table: $#ntn; field: 'id'; field: 'order_field')
	
```

! Тут используется третий (num | str) элемент в параметре key в listSort —> наверное это в .37 версии только появилось.

**PS:** Этот код не рассчитан на таблицу с миллионами строк в которой включена автоматическая разбивка на страницы. В этом случае лучше использовать колонку сортировки и по кнопке перезагрузить таблицу используя linkToTable.

**Reply by horyachka** [reply]
В моем проекте таблица более сложная.
1. Первый шаг после загрузки страницы таблицы проекта - нажатие кнопки для выбора клиентов > и загрузились клиенты.
2. Второй шаг - сортировка по дате или количеству. Для него и нужна кнопка.

Обновлю версию, добавлю код и все получится!
Спасибо за быстрые ответы!
